kernel/wait_queue/src/lib.rs

```diff
@@ -198,23 +199,27 @@ impl WaitQueue {
         // (4) Release the lock on the waitqueue.
 
         let mut wq_locked = self.0.lock();
-        let tref = if let Some(ttw) = task_to_wakeup {
-            // find a specific task to wake up
-            let index = wq_locked.iter().position(|t| t == ttw);
-            index.and_then(|i| wq_locked.remove(i))
-        } else {
-            // just wake up the first task
-            wq_locked.pop_front()
-        };
-
-        // trace!("  notify: chose task to wakeup: {:?}", tref);
-        if let Some(t) = tref {
-            // trace!("WaitQueue::notify():  unblocked task on waitqueue\n    --> WQ: {:?}", &*wq_locked);
-            t.unblock();
-            true
-        } else {
-            // trace!("WaitQueue::notify():  did nothing");
-            false
+        
+        loop {
```

Comment:
i think this could cause an infinite loop; any options to avoid that problem?

Reply:
I don't think it could because on every loop an item gets removed from `wq_locked` and so eventually `wq_locked` will run out of items and the function will return `false`.

Reply:
good enough for me :smile: